Mathlib/Control/LawfulFix.lean

```diff
@@ -225,17 +227,19 @@ variable [(x y : _) → OmegaCompletePartialOrder <| γ x y]
 
 open OmegaCompletePartialOrder.Chain
 
-theorem continuous_curry : Continuous <| monotoneCurry α β γ := fun c ↦ by
-  ext x y
-  dsimp [curry, ωSup]
-  rw [map_comp, map_comp]
-  rfl
+theorem continuous_curry : ωScottContinuous <| monotoneCurry α β γ :=
+  ωScottContinuous_iff_monotone_map_ωSup.mpr ⟨OrderHom.monotone (monotoneCurry α β γ), fun c ↦ by
+    ext x y
+    dsimp [curry, ωSup]
+    rw [map_comp, map_comp]
+    rfl⟩
 
-theorem continuous_uncurry : Continuous <| monotoneUncurry α β γ := fun c ↦ by
-  ext ⟨x, y⟩
-  dsimp [uncurry, ωSup]
-  rw [map_comp, map_comp]
-  rfl
+theorem continuous_uncurry : ωScottContinuous <| monotoneUncurry α β γ :=
+  ωScottContinuous_iff_monotone_map_ωSup.mpr ⟨fun ⦃a b⦄ a i ↦ a i.fst i.snd, fun c ↦ by
```

Comment:
This proof got quite a bit uglier, why is that?

Reply:
1. We didn't previously have to prove monotonicity since `monotoneUncurry` is an `OrderHom`
2. We need `ωScottContinuous_iff_monotone_map_ωSup` to get back to the previous definition in terms of preserving `ωSup`.

Reply:
Right, but with this suggestion https://github.com/leanprover-community/mathlib4/pull/13201#discussion_r1693712230 you can avoid re-proving monotonicity, and we should end up with something nice and clean again

Reply:
Sure, but it feels a bit idiosyncratic to order homs

Reply:
Done, thanks.